numpy/core/src/multiarray/buffer.c

```diff
@@ -953,8 +953,11 @@ _descriptor_from_pep3118_format(char *s)
     Py_DECREF(str);
     Py_DECREF(_numpy_internal);
     if (descr == NULL) {
+        PyObject *exc, *val, *tb;
+        PyErr_Fetch(&exc, &val, &tb);
         PyErr_Format(PyExc_ValueError,
                      "'%s' is not a valid PEP 3118 buffer format string", buf);
+        PyArray_ChainExceptionsCause(exc, val, tb);
```

Comment:
Looks like the prototype for ` `needs to be in an include file, it is defined in `arrayobject.c`.

Reply:
Which include file would you recommend? `npy_compat.h`?